Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -0,0 +1,1098 @@
+#include "WarpX.H"
+#include <AMReX_Scan.H>
+#include <AMReX_iMultiFab.H>
+#include <AMReX_MultiFab.H>
+
+/**
+* \brief auxiliary function to count the amount of faces which still need to be extended
+*/
+amrex::Array1D<int, 0, 2>
+WarpX::CountExtFaces() {
+    amrex::Array1D<int, 0, 2> sums{0, 0, 0};
+#ifdef AMREX_USE_EB
+
+    for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+        amrex::ReduceOps<amrex::ReduceOpSum> reduce_ops;
+        amrex::ReduceData<int> reduce_data(reduce_ops);
+        for (amrex::MFIter mfi(*m_flag_ext_face[maxLevel()][idim]); mfi.isValid(); ++mfi) {
+            amrex::Box const &box = mfi.validbox();
+            auto const &flag_ext_face = m_flag_ext_face[maxLevel()][idim]->array(mfi);
+            reduce_ops.eval(box, reduce_data,
+                [=] AMREX_GPU_DEVICE(int i, int j, int k) -> amrex::GpuTuple<int> {
+                    return flag_ext_face(i, j, k);
+                });
+        }
+
+        auto r = reduce_data.value();
+        sums(idim) = amrex::get<0>(r);
+    }
+
+    amrex::ParallelDescriptor::ReduceIntSum(&(sums(0)), AMREX_SPACEDIM);
+#endif
+    return sums;
+}
+
+
+/**
+* \brief This is the main function computing the cell extension. Where possible it computes one-way
+ *       extensions and, when this is not possible, it does eight-ways extensions.
+*/
+void
+WarpX::ComputeFaceExtensions(){
+#ifdef AMREX_USE_EB
+    amrex::Array1D<int, 0, 2> N_ext_faces = CountExtFaces();
+    amrex::ParallelDescriptor::ReduceIntSum(&(N_ext_faces(0)), 3);
+    amrex::Print()<< "Faces to be extended in x:\t" << N_ext_faces(0) <<std::endl;
```

Comment:
Are those potentially leftover debug statements?

Reply:
Yes and no. Given that for certain geometries this algorithm can fail, it's useful to know what's going on in this extension phase. For this reason, I thought it was good to have these prints always. Anyways, if you think it's better, I can add an #ifdef AMREX_DEBUG.

Reply:
Yes, that makes. You can also rely on the `WarpX::verbose` option to control it at runtime instead